Upgrade to Webpack 2 for Treeshaking and Codesplitting

zacck-zz opened the pull request:
#### What does this PR do?
Upgrade the module bundler version to webpack 2.
#### Description of Task to be completed?
I upgraded the package version of  webpack and webpack-dev-server
I updated the webpack configuration to work with webpack 2 standards
#### How should this be manually tested?
Currently you can run yarn dev / npm dev to see  the app compiled and running
#### Any background context you want to provide?
I upgraded to webpack 2 to enable codesplitting and treeshaking as the generated bundle grows larger [webpack 2](https://medium.com/webpack/webpack-2-and-beyond-40520af9067f)

zacck committed:
work on upgrading to webpack 2 for codesplitting features and treeshaking

waiyaki commented on webpack.config.js:
```diff
@@ -18,17 +18,17 @@ const config = {
       {
         test: /\.scss$/,
-        loader : 'style!css!sass!postcss'
+        loader : 'style-loader!css-loader!sass-loader!postcss-loader'
```
Now that we're switching to webpack 2, let's switch this config to the webpack 2 specific way [using `use` instead of `loader`](https://webpack.js.org/guides/migrating/#module-loaders-is-now-module-rules).

waiyaki reviewed (changes requested):
Great work @zacck! I love it! 💯 

- Could you also kindly remove `new webpack.optimize.DedupePlugin()` from the config, now that it was removed from webpack 2?
- Also kindly edit how loaders are defined. I've linked to webpack docs in the specific comment.

Thanks! 👍

zacck committed:
work on using webpack 2 style loaders

zacck-zz commented:
@waiyaki aweh!  So I removed the dedupe plugin and also I changed the chaining of loaders to go webpack 2 style still a bit more to do on this config but more as I understand the app by and by.

waiyaki commented on tests/TestRunner.elm:
```diff
@@ -3,11 +3,12 @@ port module TestRunner exposing (..)
 import Tests
 import Test.Runner.Node exposing (run, TestProgram)
 import Json.Encode exposing (Value)
+import Sample
```
This is making the [tests fail](https://travis-ci.org/AndelaOSP/shufflebox-frontend/jobs/219171442). Maybe you committed it by mistake? 🙂

zacck-zz replied:
Ah apologies thats a leaked  experiment from my testing PR  that should not necessarily be linked to this.

zacck-zz replied:
Tested sorry about that should be back to passing by now

waiyaki replied:
Thank you for doing this. 🥇

zacck committed:
remove erronous import

zacck committed:
remove erronous test

waiyaki commented on webpack.config.js:
```diff
@@ -15,28 +15,34 @@ const config = {
       {
         test: /\.scss$/,
-        loader : 'style!css!sass!postcss'
+        use : ['style-loader','css-loader', 'sass-loader', 'postcss-loader']
```
This is a little pedantic and I realize that we don't have a JavaScript linter 😓 .
Could you kindly adjust the spacing so that it's consistent? Goes for the `elm` loaders as well. 🙂 

The rest of it looks good to me. 👍

zacck committed:
work on spacing in webpack.config.js

zacck-zz commented:
@waiyaki no worries done did.

waiyaki commented:
@zacck Merged. Thanks a bunch for this! 🎉 🥇
